### DISCUSSION TITLE
URL response save in UTF-8

### DISCUSSION THREAD

**Discussion by Kolhun** [curl/curl#11768] [Q&A] [answered]
Good afternoon, I wrote here earlier but did not find the necessary information. They receive the server's response via a POST request, how can the result be saved in UTF-8 (Cyrillic)? jq did not help, further work with the undecoded string causes problems

![image](https://github.com/curl/curl/assets/36815965/5f03fbd7-2fad-4e5d-a03d-389a0a8747e3)

**Comment by dfandrich** [comment]
I'm not sure exactly what you're asking. Are you looking for a way to convert from one character set to another? iconv can do that. jq can be first used to extract the text to convert from a JSON block before conversion. This has nothing to do with curl, though.

**Comment by jzakrzewski** [comment, marked as answer]
Well, in JSON Unicode characters from outside of ASCII range are represented with escape sequences. Nothing, that saves valid JSON will convert that to the actual cyrillic characters.

However, jq actually can do this, and does by default:
```
$ echo '{"g": "\u0430\u043b"}' | jq > /tmp/file.txt
$ cat /tmp/file.txt
{
  "g": "ал"
}
$ hexdump -C /tmp/file.txt
00000000  7b 0a 20 20 22 67 22 3a  20 22 d0 b0 d0 bb 22 0a  |{.  "g": "....".|
00000010  7d 0a                                             |}.|
00000012
$ 
```
(And please, for the love of Cthulhu, use text instead of images...)
